sdk/src/azure/core/az_precondition.c

```diff
@@ -2,12 +2,15 @@
 // SPDX-License-Identifier: MIT
 
 #include <azure/core/internal/az_precondition_internal.h>
+
+#include <assert.h>
```

Comment:
Is `assert.h` always available, even on IoT devices (a no-op is fine) and C99 compatible?

Reply:
It is part of the standard, and it is a macro. Nothing prevents the platform compiler to define it as no-op.